### PR TITLE
merge some upgrade tests

### PR DESCRIPTION
**Pull request by phoracek**
Signed-off-by: Petr Horacek <phoracek@redhat.com>

**What this PR does / why we need it**:

For each release, we test upgrades from every previous release to the
newly created one. That takes a lot of time and is obviously getting
only worse.

This patch merges two of the three upgrade tests into a single one. They
are both just checking the state of the cluster, so we are not risking
too much by merging them.

This should save us 30 % of the time on the suite.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/lifecycle/upgrade_test.go [hunk 1]
```diff
@@ -42,11 +42,11 @@ var _ = Context("Cluster Network Addons Operator", func() {
...
 				})
 
-				It("it should report expected deployed container images", func() {
+				It("it should report expected deployed container images and leave no leftovers from the previous version", func() {
+					By("Checking reported container images")
 					CheckReleaseUsesExpectedContainerImages(newRelease)
-				})
 
-				It("should run with no leftovers from the original version", func() {
+					By("Checking for leftover objects from the previous version")
```
Comment (qinqon) [inline test/e2e/lifecycle/upgrade_test.go:49]: Can we Sprintf the previous version ?

Comment (phoracek) [reply test/e2e/lifecycle/upgrade_test.go:49]: Done in the context

**Comment by phoracek** [comment]
/override pull-e2e-cluster-network-addons-operator-workflow-k8s

**Pull request merged**